libraries/consul_installation_binary.rb

```diff
@@ -59,6 +59,15 @@ def action_create
             to ::File.join(options[:extract_to], new_resource.version, 'consul')
             not_if { windows? }
           end
+
+          link "#{node.config_prefix_path}\\consul.exe" do
+            to ::File.join(options[:extract_to], new_resource.version, 'consul.exe')
+            only_if { windows? }
+          end
+          windows_path node.config_prefix_path do
```

Comment:
Please, try if we can just use `env` instead:
```
  env 'path' do
    action :modify
    delim ::File::PATH_SEPARATOR
    value node.config_prefix_path
    only_if { windows? }
  end
```
It should work as expected, on Chef >= 12, without having to add a `windows` dependency

Reply:
I can certainly take a look.

But the `windows` cookbook isn't a new dependency: the `consul` cookbook depends on `build-essential`, which depends on `seven-zip`, which has a dependency on `windows`. So this is more making it an explicit dependency. And with `windows_path` becoming part of core Chef, it seems more clear going forward.

Let me know what you think.

Reply:
> the `consul` cookbook depends on `build-essential`, which depends on `seven-zip`, which has a dependency on `windows`

Ah, sorry, I forgot about that. Then I'm OK with setting that dependency explicitly. 👍

Reply:
No worries, I had the "wait, why did that work" moment myself :)